osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -126,11 +126,19 @@ public void PrepareTrackForPreviewLooping()
         }
 
         /// <summary>
-        /// Transfer a valid audio track into this working beatmap. Used as an optimisation to avoid reload / track swap
-        /// across difficulties in the same beatmap set.
+        /// Attempts to transfer the audio track to a target working beatmap, if valid for transferring.
+        /// Used as an optimisation to avoid reload / track swap across difficulties in the same beatmap set.
         /// </summary>
-        /// <param name="track">The track to transfer.</param>
-        public void TransferTrack([NotNull] Track track) => this.track = track ?? throw new ArgumentNullException(nameof(track));
+        /// <param name="target">The target working beatmap to transfer this track to.</param>
+        /// <returns>Whether the track has been transferred to the <paramref name="target"/>.</returns>
+        public virtual bool TryTransferTrack([NotNull] WorkingBeatmap target)
+        {
+            if (BeatmapInfo?.AudioEquals(target.BeatmapInfo) != true || Track.IsDummyDevice)
```

Comment:
I see that a17eed64 has sparked a change from `track` to `Track`, to ensure that the track has loaded - but I'm not sure that this is the right way to go here - would it not make sense to just *not* transfer the track if it's not loaded, and let the next `WorkingBeatmap` do it?

Reply:
The way I'm seeing it here, and what `MusicController` looks to have been doing prior to this PR, is that if we are requested for transferring a track, we can't possibly be in a non-loaded state, and we should definitely throw instead of not doing anything.

Reply:
So it was just a safety change on your end? It wasn't sparked by any actual issues after the other changes?

Reply:
Yeah, noticed it while looking at the other suggestions.